File: spring-mvc-plain-java-example/src/main/java/de/cfranzen/examples/webflux/CustomerRepository.java
package de.cfranzen.examples.webflux;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.stream.Stream;

@Repository
public interface CustomerRepository extends CrudRepository<Customer, Long> {

    Stream<Customer> findByFirstNameAndLastName(String firstName, String lastName);
}

File: spring-webflux-kotlin-example/src/main/kotlin/de/cfranzen/examples/webflux/kotlin/CustomerRepository.kt
package de.cfranzen.examples.webflux.kotlin

import kotlinx.coroutines.flow.Flow
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import org.springframework.stereotype.Repository

@Repository
interface CustomerRepository : CoroutineCrudRepository<Customer, Long> {

    fun findByFirstNameAndLastName(firstName: String, lastName: String): Flow<Customer>
}
